### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by a-b**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          path: dist
+          merge-multiple: true
+
+      - name: Create GitHub Release
+        run: |
+          echo "# Changes in this release" > release_notes.md
+
+          draft_flag="${{ github.event.inputs.draft || 'true' }}"
+
+          gh release create "${{ needs.prepare-release.outputs.release_version }}" \
```
Comment (Samze) [inline .github/workflows/release.yml:142]: **question** Does this also create the tag?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
+name: Release
+
+on:
+  workflow_dispatch:
```
Comment (Samze) [inline .github/workflows/release.yml:4]: *nit/non-blocking* We should only release if unit tests have passed

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+      - name: Build Binary
+        env:
+          GOOS: ${{ matrix.goos }}
+          GOARCH: ${{ matrix.goarch }}
+          CGO_ENABLED: 0
+        run: |
+          output_name="stack-auditor-${{ matrix.os }}-${{ matrix.suffix }}"
+          echo "::group::Building for ${{ matrix.os }}-${{ matrix.arch }}"
+          go build -v -trimpath -ldflags="-s -w" -o "dist/$output_name" .
```
[the remark this answers is not in the document]
Comment (a-b) [reply .github/workflows/release.yml:110]: Good catch - thanks.

File: .github/workflows/release.yml (hunk 1 above)
Comment (a-b) [reply .github/workflows/release.yml:142]: I don't think so - good point

File: .github/workflows/release.yml (hunk 2 above)
Comment (a-b) [reply .github/workflows/release.yml:4]: I rewrote the whole unit pipeline, but I considered adding it via a separate PR.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Samze) [reply .github/workflows/release.yml:154]: Can you speak more to this? When will the tag get created, as part of the publishing of the draft release?

Comment (a-b) [reply .github/workflows/release.yml:154]: Sure, it's [here](https://github.com/cloudfoundry/stack-auditor/pull/83/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R135-R140)

Comment (a-b) [reply .github/workflows/release.yml:154]: It's a part of the release process before the draft.

**Review by Samze — APPROVED** [review]

**Pull request merged**